ADEN-4013 Recover ad video in Ooyala player.

drets opened the pull request.

Commit:
ADEN-4013 Temporary replace Ooyala player to static file and force recovery

Commit:
ADEN-4013 next changes

Commit:
ADEN-4013 Use separated ima library

Commit:
ADEN-4013 Use Ooyala files from cdn

drets committed:
ADEN-4013 Tweaking global IMA vars

• mock unshift for on failure callbacks
• loading IMA lib manually
• call on success callbacks

Commit:
ADEN-4013 refactor

drets committed:
ADEN-4013 Minor clean-up.

drets committed:
ADEN-4013 Remove redundant blocking check.

drets committed:
ADEN-4013 Small refactoring.

WikiaTech commented:
SonarQube analysis reported 2 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Ooyala.js#L78](https://github.com/Wikia/app/blob/7490fcbef4c3e9004f45ca8fb6100511f041f641/extensions/wikia/VideoHandlers/js/handlers/Ooyala.js#L78): Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented on extensions/wikia/ARecoveryEngine/js/helper.js:
```diff
@@ -67,7 +67,7 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
 					xmlHttp.send();
 				} catch (e) {}
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEmptyBlock)

fraszczakszymon commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,94 @@ define('wikia.videohandler.ooyala', [
 		}
 
+		function initPlayer() {
+			if (recoveryHelper.isBlocking()) {
```
&& isRecoveryEnabled

drets committed:
ADEN-4013 Run player on recovery ima lib fail.

WikiaTech commented:
SonarQube analysis reported 2 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Ooyala.js#L78](https://github.com/Wikia/app/blob/c09720b18604d2dca5544a566642dae9a8898831/extensions/wikia/VideoHandlers/js/handlers/Ooyala.js#L78): Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

fraszczakszymon commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,104 @@ define('wikia.videohandler.ooyala', [
+
+			createParams.vast.tagUrl = recoveryHelper.getSafeUri(createParams.vast.tagUrl);
+
+			loadJs(ima3LibUrl).done(function () {
```
JSLint: loadJs was used before it was defined

fraszczakszymon commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,104 @@ define('wikia.videohandler.ooyala', [
+		}
+
+		// Only load the Ooyala player code once, Ooyala AgeGates will break if we load this asset more than once.
+		if (win.OO !== undefined) {
```
Is it correct?

drets replied:
We understand why it is here (thanks for the comment above code). 
Let's left it here in order to keep the same behavior.

fraszczakszymon commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,104 @@ define('wikia.videohandler.ooyala', [
+			log('Create recovered Ooyala player', log.levels.info, logGroup);
+
+			win.googleImaSdkFailedCbList = {
+				originalCbList: [],
```
Are you sure that it is a regular array? Not kind of lazy queue?

edit: maybe we can clear this array instead of overriding it

drets replied:
> Are you sure that it is a regular array? Not kind of lazy queue?

Yes, it is a regular array.

> edit: maybe we can clear this array instead of overriding it

Nope, since the callbacks list is filled _and_ executed inside IMA plugin.

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/helper.js:
```diff
@@ -81,12 +81,31 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
+	}
+
+	function runAfterDetection(callback) {
+		if (isSourcePointResultDefined()) {
```
You can achieve the same thing without checking current status of detection. `addOnBlockingCallback` is using lazy queue and for sure it's fired once. It would be beneficial to add second queue for `sp.not_blocking` and reuse it in the same way

fraszczakszymon commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -23,9 +24,10 @@ define('wikia.videohandler.ooyala', [
 	 */
 	return function (params, vb) {
-		var containerId = vb.timeStampId(params.playerId),
+		var ima3LibUrl = 'https://imasdk.googleapis.com/js/sdkloader/ima3.js',
```
https seems not needed, use `//imasdk.goo...` please

drets committed:
ADEN-4013 CR fixes.

• makeQueue returns queue
• add unit test for lazyQueue
• add on not blocking event queue
• etc

WikiaTech commented:
SonarQube analysis reported 4 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L22](https://github.com/Wikia/app/blob/1c2367212f4dcb20f90709b6e05099510f5b5621/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L103](https://github.com/Wikia/app/blob/1c2367212f4dcb20f90709b6e05099510f5b5621/extensions/wikia/AdEngine/js/run/desktop.run.js#L103): Function has 10 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Ooyala.js#L78](https://github.com/Wikia/app/blob/1c2367212f4dcb20f90709b6e05099510f5b5621/extensions/wikia/VideoHandlers/js/handlers/Ooyala.js#L78): Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented:
SonarQube analysis reported 4 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L22](https://github.com/Wikia/app/blob/dc4e3a4c985c1f76d476455e9d6b6d501be4010b/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L103](https://github.com/Wikia/app/blob/dc4e3a4c985c1f76d476455e9d6b6d501be4010b/extensions/wikia/AdEngine/js/run/desktop.run.js#L103): Function has 10 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Ooyala.js#L78](https://github.com/Wikia/app/blob/dc4e3a4c985c1f76d476455e9d6b6d501be4010b/extensions/wikia/VideoHandlers/js/handlers/Ooyala.js#L78): Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

bognix commented on extensions/wikia/ARecoveryEngine/js/helper.js:
```diff
@@ -67,7 +70,7 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
 					xmlHttp.send();
 				} catch (e) {}
```
address the comment above. If error happens consider logging it at least so it doesn't fail silently

bognix commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,97 @@ define('wikia.videohandler.ooyala', [
+					fn();
+				}, win.OO);
+			}).fail(function() {
+				log('Recovered ima3 lib is failded to load', log.levels.info, logGroup);
```
remove `is` and change `failded` to `failed`

bognix commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -79,55 +81,97 @@ define('wikia.videohandler.ooyala', [
+
+			win.googleImaSdkFailedCbList = {
+				originalCbList: [],
+				unshift: function (cb) {
```
there is a difference between `unshift` and `push`. Why the property is called `unshift` but `push` is called inside?

bognix commented:
Consider addressing `MAJOR Ooyala.js#L78: Remove this commented out code` from Sonar

bognix reviewed (approved):
Few minor comments left.

LGTM

drets committed:
ADEN-4013 CR fixes 2.

WikiaTech commented:
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L22](https://github.com/Wikia/app/blob/d8124180aaa8e1bd24a5575ceec53e08a6f6880f/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L103](https://github.com/Wikia/app/blob/d8124180aaa8e1bd24a5575ceec53e08a6f6880f/extensions/wikia/AdEngine/js/run/desktop.run.js#L103): Function has 10 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

drets committed:
Merge branch 'dev' of github.com:Wikia/app into ADEN-4013

WikiaTech commented:
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L22](https://github.com/Wikia/app/blob/cb82de1e5ca0ddbee460c3f4dbc6a6e4837c33b7/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L104](https://github.com/Wikia/app/blob/cb82de1e5ca0ddbee460c3f4dbc6a6e4837c33b7/extensions/wikia/AdEngine/js/run/desktop.run.js#L104): Function has 11 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

drets committed:
ADEN-4013 Remove trailing spaces.

WikiaTech commented:
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L22](https://github.com/Wikia/app/blob/2d985db19887f9a57b6547a114ca16a85b5ef15b/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [desktop.run.js#L104](https://github.com/Wikia/app/blob/2d985db19887f9a57b6547a114ca16a85b5ef15b/extensions/wikia/AdEngine/js/run/desktop.run.js#L104): Function has 11 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

bognix commented:
LGTM
